### ISSUE TITLE
wgan_gp d_steps repetitions use the same input images.

### ISSUE THREAD

**Issue by JosiahKane** [keras-team/keras-io#330]
In the [wgan_gp example code](https://github.com/keras-team/keras-io/blob/master/examples/generative/wgan_gp.py), the discriminator is trained 3 times per train step, and the generator 1. However, each of these 3 training steps uses the same real image batch. Is this a deliberate decision?

For contrast in the [reference implementation](https://github.com/igul222/improved_wgan_training/blob/master/gan_mnist.py#L225)  there are `disc_iters` repetitions of getting fresh data from the data source to train against. This would more uniformly train the discriminator (critic) both in sampling the gradients for enforcing the 1-Lipschitz constraint and in promoting a high score for that particular batch.

**Comment by pcoet** [comment]
Hi @JosiahKane, thanks for getting in touch. This question would be a better fit for one of the forums below. Can you repost? Thanks!

* [TensorFlow forum](https://discuss.tensorflow.org/)
* [Keras mailing list](https://groups.google.com/forum/#!forum/keras-users)
